src/git/store.mli

```diff
@@ -0,0 +1,737 @@
+      The list returned does not contain all git object available in your
+      repository but only which {i loosed} one. *)
+
+  val read_p : ztmp:Cstruct.t ->
+    dtmp:Cstruct.t ->
+    raw:Cstruct.t ->
+    window:Inflate.window ->
+    state -> Hash.t -> (t, error) result Lwt.t
+  (** [read_p ~ztmp ~dtmp ~raw ~window state hash] can retrieve a git {i loose}
+      object from the file-system. It de-serializes the git object to an OCaml
+      value. This function needs some buffers:
+
+      {ul
+      {- [window] is a buffer used by the {!Inflate} module}
+      {- [ztmp] is a buffer used to store the inflated flow}
+      {- [dtmp] is a buffer used by the decoder to save the inflated flow (and
+      keep it for an alteration)}
+      {- [raw] is a buffer used to store the input flow}}
+
+      This function can be used in a concurrency context only if the specified
+      buffers are not used by another process. This function does not allocate
+      any buffer and uses only the specified buffers to construct the OCaml
+      value. The git object returned respects the predicate [digest(result) =
+      hash]. Otherwise, we return an {!error}:
+
+      {ul
+      {- {!FileSystem.File.error} when we can not access to the git object in the
+      file-system (because it does not exist or the structure of the git
+      repository is wrong).}
+      {- {!Value.D.error} when we can not de-serialize xor inflate the requested
+      git {i loose} object. That means, the git {i loose} object does not
+      respect the Git format.}}
+  *)
+
+  val read_s : state -> Hash.t -> (t, error) result Lwt.t
+  (** [read_s state hash] is the same process than {!read_p} but we use the
+      state-defined buffer. That means the client can not use this function in a
+      concurrency context with the same [state]. *)
+
+  val read : state -> Hash.t -> (t, error) result Lwt.t
+  (** Alias of {!read_s}. *)
+
+  val size_p : ztmp:Cstruct.t ->
+    dtmp:Cstruct.t ->
+    raw:Cstruct.t ->
+    window:Inflate.window ->
+    state -> Hash.t -> (int64, error) result Lwt.t
+  (** [size_p ~ztmp ~dtmp ~raw ~window state hash] returns the size of the git
+      {i loose} object which respects the predicate [digest(object) = hash]. The
+      size is how many byte(s) are needed to store the serialized (but not
+      inflated) git object in bytes (without the header).
+
+      As {!read_p}, {!size_p} needs some buffers. The client can use this
+      function in a concurrency context when the buffers specified are not used
+      by another process.
+
+      As {!read_p}, {!size_p} can return an {!error}.
+  *)
+
+  val size_s : state -> Hash.t -> (int64, error) result Lwt.t
+  (** [size_s state hash] is the same process than {!size_p} but we use the
+      state-defined buffer. That means the client can not use this function in a
+      concurrency context with the same [state]. *)
+
+  val size : state -> Hash.t -> (int64, error) result Lwt.t
```

Comment:
I think I would just expose the top-level function (with the `_s` suffix) and move the non-blocking one (with the `_p` suffix into a sub-module).